src/EditorFeatures/Core/InlineHints/InlineHintsDataTaggerProvider.cs

```diff
@@ -114,7 +116,10 @@ protected override IEnumerable<SnapshotSpan> GetSpansToTag(ITextView? textView,
             var options = GlobalOptions.GetInlineHintsOptions(document.Project.Language);
 
             var snapshotSpan = documentSnapshotSpan.SnapshotSpan;
-            var hints = await service.GetInlineHintsAsync(document, snapshotSpan.Span.ToTextSpan(), options, cancellationToken).ConfigureAwait(false);
+            var hints = await service.GetInlineHintsAsync(
+                document, snapshotSpan.Span.ToTextSpan(), options,
+                displayAllOverride: _inlineHintKeyProcessor?.State is true,
+                cancellationToken).ConfigureAwait(false);
```

Comment:
@tmat instead of this, we woudl do a `with` on the value returned by `GlobalOptions.GetInlineHintsOptions(document.Project.Language);` and would supply this value there.

Reply:
Either way is fine. Whatever feels better for the service API.